### PR TITLE
DOC-9209: How-to Guides for Server Developers › Query › Indexes

### PR DESCRIPTION
**Pull request by simon-dew**
The following draft documentation is ready for review:

* [Creating Indexes](https://simon-dew.github.io/docs-site/DOC-9196/server/7.0-reorg/guides/create-index.html)
* [Index Placement](https://simon-dew.github.io/docs-site/DOC-9196/server/7.0-reorg/guides/place-index.html)
* [Deferring Indexes](https://simon-dew.github.io/docs-site/DOC-9196/server/7.0-reorg/guides/defer-index.html)
* [Selecting Indexes](https://simon-dew.github.io/docs-site/DOC-9196/server/7.0-reorg/guides/select-index.html)
* [Dropping Indexes](https://simon-dew.github.io/docs-site/DOC-9196/server/7.0-reorg/guides/drop-index.html)

Docs issue: [DOC-9209](https://issues.couchbase.com/browse/DOC-9209)

### CODE REVIEW

File: modules/guides/pages/create-index.adoc [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+The following example creates an unnamed primary index.
+
+[source,csharp]
+----
+include::example$index/create-index.cs[tag=primary]
+----
+
+{github}
+
+For further details, refer to https://docs.couchbase.com/sdk-api/couchbase-net-client/api/Couchbase.Management.Query.IQueryIndexManager.html[IQueryIndexManager^].
```
Comment (maria-robobug) [inline modules/guides/pages/create-index.adoc:99]: Ooh this is quite a nice idea, I'll add it to the KV examples where I can!

File: modules/guides/pages/create-index.adoc [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+
+1. Specify the keyspace on which to create the index.
+
+2. If you want to specify a name for the index:
+
+  a. Use `CreatePrimaryQueryIndexOptions` to specify the index options.
+
+  b. In the index options, use the `IndexName` method to specify the index name.
+
+'''
```
Comment (maria-robobug) [inline modules/guides/pages/create-index.adoc:81]: I like the line separation, another good idea I'll steal 👍  😄 .

**Comment by simon-dew** [comment]
- [x] In Create Indexes, move explanation of `#primary` index name out of the N1QL procedure
- [x] In Drop Indexes, consider separating **Dropping Unnamed Primary Indexes** and **Dropping Named Primary Indexes**
- [x] Broken link in Select Index page
- [x] SDKs do not currently support creating indexes on named collections!

**Comment by maria-robobug** [comment]
Added examples here @simon-dew :
- .NET: https://github.com/couchbase/docs-sdk-dotnet/pull/231 
- Nodejs: https://github.com/couchbase/docs-sdk-nodejs/pull/163
- Python: https://github.com/couchbase/docs-sdk-python/pull/179 
- Java: https://github.com/couchbase/docs-sdk-java/pull/299 

> Finally, it seems to be the case that all the SDKs (Java, .NET, Node.js, and Python provide a method for building all deferred indexes in a keyspace, but not for building individual deferred indexes. Can you confirm this is the case? 

From what I could see in the SDK API docs and code, the `buildDeferredIndexes` method will build any deferred index in the specified bucket it seems. No way to specify a single one.

> It seems there is no way of dropping a named primary index using the Java SDK. Is this true? (DropPrimaryIndex and DropPrimaryQueryIndexOptions do not provide a way to specify the index name.) https://docs.couchbase.com/sdk-api/couchbase-java-client/com/couchbase/client/java/manager/query/QueryIndexManager.html

Yeah I couldn't see any way to pass a named primary index with the Java SDK unfortunately.

**Comment by simon-dew** [comment]
I'm going to merge this now. Still waiting for @maria-robobug 's code samples to be checked and merged. If there are any further changes required following on from that, I'll make a new pull request.

**Pull request merged**